chain/client/src/client_actor.rs

```diff
@@ -1064,7 +1064,8 @@ impl ClientActor {
                         &mut self.client.chain,
                         highest_height,
                         &self.network_info.most_weight_peers
-                    ))
+                    ));
+                    false
```

Comment:
It's already not calling state sync if header sync is too far (the if statement above this clause).
This call to block sync returns true if state sync is needed. Right now you will never return true.

Reply:
++